src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecProc.java

```diff
@@ -86,7 +87,14 @@ public int join() throws IOException, InterruptedException {
             LOGGER.log(Level.FINEST, "Waiting for websocket to close on command finish ({0})", finished);
             finished.await();
             LOGGER.log(Level.FINEST, "Command is finished ({0})", finished);
-            return watch.exitCode().join();
+
+            CompletableFuture<Integer> exitCodeFuture = watch.exitCode();
+            if (exitCodeFuture == null) {
+                LOGGER.log(Level.FINEST, "Watcher return 'null' instead of exitCode");
```

Comment:
Maybe we should extend https://github.com/jenkinsci/kubernetes-plugin/blob/294ad4250b9169313db589c44fe993f5191e7ce8/src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java#L563-L567 to pass the `printStream` to `proc` so unusual issues can be printed in the build log, rather than the system log where they would only be visible to admins.

Reply:
Good idea, but I don't know how to do it in a better way. Can I add an additional arg for `ContainerExecProc` to pass `printStream` and then use it to print a log?

Reply:
I think it can be implemented as a separate PR to unblock the issue fix. I can prepare a PR for that.

Reply:
> Can I add an additional arg

That is what I was thinking.

> I think it can be implemented as a separate PR

Agreed, it was just an idea.